### PR TITLE
[chore] Add a Docker-based development server.

### PR DESCRIPTION
**Pull request by lukesneeringer**
This commit adds a Docker-based "development server" (just a thin wrapper around the Jekyll server), and expands `CONTRIBUTING.md` with instructions on using it.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,7 +1,57 @@
...
+## Development Environment
+
+If you are contributing documentation (rather than samples) and want to be able
+to view it in your browser, the easiest way to do so is to run the provided
+development server.
+
+We use [GitHub Pages][1] to make this documentation available, which uses
+[Jekyll][2] under the hood.
+
+If you have [Docker][3] installed, simply clone this repository and run the
```
Comment (timburks) [inline CONTRIBUTING.md]: Minor: I'd like to remove the word "simply" here. [This](https://www.knowledgeowl.com/home/dont-say-simply-jim-fisher) more-or-less explains why.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,7 +1,57 @@
...
+`serve.sh` file at the root of the repository. This script does two things:
+
+  * It builds the provided Docker image (unless you already have it) and tags
+    it as `googleapis-site`.
+  * It runs the `googleapis-site` image.
+
+The Jekyll development server uses port 4000 by default, so simply point your
```
Comment (timburks) [inline CONTRIBUTING.md]: I think we can delete this word with no loss of meaning.

File: CONTRIBUTING.md (hunk 1 above)
Comment (lukesneeringer) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 2 above)
Comment (lukesneeringer) [reply CONTRIBUTING.md]: Done.

**Pull request merged**